sale_loyalty_limit/models/sale_order.py

```diff
@@ -0,0 +1,103 @@
+                (
+                    "commercial_partner_id",
+                    "=",
+                    self.commercial_partner_id.id,
+                ),
+            ]
+            order_count = self.env["sale.order"].search_count(customer_domain)
+            if order_count < rule.rule_max_customer_application:
+                limit_customer_reached = False
+        if limit_customer_reached and not coupon:
+            return {
+                "error": _(
+                    "This promotion was already applied %(count)s times "
+                    "for this customer and there's an established limit "
+                    "of %(max)s."
+                )
+                % {
+                    "count": order_count,
+                    "max": rule.rule_max_customer_application,
+                }
+            }
+        elif limit_customer_reached and coupon:
+            return {
+                "error": _(
+                    "This promo code was already applied %(count)s times for this "
+                    "customer and there's an stablished limit of %(max)s for this "
+                    "promotion."
+                )
+                % {
+                    "count": order_count,
+                    "max": rule.rule_max_customer_application,
+                }
+            }
+        return None
+
+    def _check_salesmen_limit(self, program):
+        limit_salesmen_reached = True
+        strict_limit = True
+        for rule in program.rule_ids:
+            salesman_rule = rule.rule_salesmen_limit_ids.filtered(
+                lambda x: self.user_id == x.rule_user_id
+            )
+            max_rule = salesman_rule.rule_max_salesman_application
+            times_used = salesman_rule.rule_times_used
+            if not times_used or times_used < max_rule:
+                limit_salesmen_reached = False
+            if not rule.rule_salesmen_strict_limit:
+                strict_limit = False
+        if limit_salesmen_reached:
+            return {
+                "error": _(
+                    "This promo code was already applied %(times)s times "
+                    "for this salesman and there's an established limit "
+                    "of %(max)s for this promotion."
+                )
+                % {"times": times_used, "max": max_rule}
+            }
+        if strict_limit and not salesman_rule:
+            return {"error": _("This promotion is restricted to the listed salesmen.")}
+        return None
+
+    def _program_check_compute_points(self, programs):
+        res = super()._program_check_compute_points(programs)
+        # Iterate through the programs that initially have no errors
+        for program, result in res.items():
+            if result == "error":
+                continue
+            # Customer limit rules
+            if all(rule.rule_max_customer_application for rule in program.rule_ids):
+                error = self._check_customer_limit(program)
```

Comment:
This isn't the logic we need. If only one rule in the program is valid, the program is valid and we break the loop. The same with the customer limit